membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -122,178 +108,126 @@ case object Years extends UpToPeriodsType {
-
-  def get(jsonString: String): Option[FrontendId] =
-    all.find(_.name == jsonString)
+  def totalChargesMinorUnit: Int =
+    ratePlanCharges.map(c => (c.pricing.prices.head.amount * 100).toInt).list.toList.sum
+
+  def chargesPrice: PricingSummary =
+    ratePlanCharges
+      .map(_.pricing)
+      .list
+      .toList
+      .reduce((f1, f2) =>
+        PricingSummary(
+          f1.underlying.keySet
+            .intersect(f2.underlying.keySet)
+            .map { currency =>
+              currency -> f1.underlying(currency).+(f2.underlying(currency))
+            }
+            .toMap,
+        ),
+      )
+
+  def billingPeriod: Validation[String, BillingPeriod] = {
+    val billingPeriods = ratePlanCharges.list.toList.map(c => c.billingPeriod).distinct
+    billingPeriods match {
+      case Nil => Validation.f[BillingPeriod]("No billing period found")
+      case b :: Nil => b
+      case _ => Validation.f[BillingPeriod]("Too many billing periods found")
+    }
+  }
 
-}
+  private def productRatePlan(catalog: Catalog) = {
+    catalog.productRatePlans(productRatePlanId)
+  }
 
-case class Catalog(
-    map: Map[ProductRatePlanId, CatalogZuoraPlan],
-)
+  def product(catalog: Catalog): Product =
+    catalog.products.getOrElse(productRatePlan(catalog).productId, Product.UnknownProduct)
 
-/** A higher level representation of a number of Zuora rate plan charges
-  */
-sealed trait RatePlanChargeList {
-  def benefits: NonEmptyList[Benefit]
-  def currencies: Set[Currency] = price.currencies
-  def billingPeriod: BillingPeriod
-  def price: PricingSummary
-  def subRatePlanChargeId: SubscriptionRatePlanChargeId
-}
+  def name(catalog: Catalog): String =
+    productRatePlan(catalog).name
 
-/** Same as above but we must have exactly one charge, giving us exactly one benefit This is used for supporter, partner, patron and digital pack subs
-  */
-case class RatePlanCharge[+B <: Benefit, +BP <: BillingPeriod](
-    benefit: B,
-    billingPeriod: BP,
-    price: PricingSummary,
-    chargeId: ProductRatePlanChargeId,
-    subRatePlanChargeId: SubscriptionRatePlanChargeId,
-) extends RatePlanChargeList {
-  def benefits = NonEmptyList(benefit)
-}
+  def productType(catalog: Catalog): ProductType =
+    productRatePlan(catalog).productType
 
-/** Paper plans will have lots of rate plan charges, but the general structure of them is that they'll give you the paper on a bunch of days, and if
-  * you're on a plus plan you'll have a digipack
-  */
-case class PaperCharges(dayPrices: Map[PaperDay, PricingSummary], digipack: Option[PricingSummary]) extends RatePlanChargeList {
-  def benefits = NonEmptyList.fromSeq[Benefit](dayPrices.keys.head, dayPrices.keys.tail.toSeq ++ digipack.map(_ => Digipack))
-  def price: PricingSummary = (dayPrices.values.toSeq ++ digipack.toSeq).reduce(_ |+| _)
-  override def billingPeriod: BillingPeriod = BillingPeriod.Month
```

Comment:
actually not all papers are monthly, I found some migrated in Echo Legacy that are still running and are actually quarterly